### DISCUSSION TITLE
Import (export?) additional GIS formats (using existing GDAL drivers)

### DISCUSSION THREAD

**Discussion by swiss-knight** [domlysz/BlenderGIS#879] [Ideas]
Hello 👋

currently (2024-04-19), BlenderGIS relies on those four formats for importing GIS data:

![image](https://github.com/domlysz/BlenderGIS/assets/19967599/69ce9abf-2fec-4f94-903b-5629a1fab074)

I'd like to know if it would be possible to make BlenderGIS importing other GIS formats.

It's not about reinventing the weel, **[GDAL](https://github.com/OSGeo/gdal)** already has many drivers for that:
- vector: https://gdal.org/drivers/vector/index.html
- raster: https://gdal.org/drivers/raster/index.html

So basically, the idea would be to rely on GDAL for that import(/export?) task.

To begin with, it would be nice to be able to import the most frequent formats, such as e.g. for vector data, most of those being standardized by the [Open Geospatial Consortium](https://www.ogc.org/) (a.k.a. OGC):
- local files: [GeoPackage](https://www.geopackage.org/), [GeoJSON](https://geojson.org/), [GML](https://www.ogc.org/standard/gml/), [KML](https://www.ogc.org/standard/kml/), [GPX](https://www.topografix.com/gpx.asp),
- database data: [PostGIS](https://postgis.net/) tables
- or even data from web services: [WFS](https://www.ogc.org/standard/wfs/), [OGC API Features](https://ogcapi.ogc.org/features/)

Thanks! 🙏

By the way, very nice work done on BlenderGIS, which bridges the gap between 2D spatial data and true 3D data manipulation and visualization in the GIS world!

**Comment by dcb-l** [comment]
Would love to see this! I'd add FlatGeobuf to the initial wish list for supported files due to its performance.

**Comment by srschd** [comment]
Is anyone already working on this topic? I would be very interested, because BlenderGIS and BlenderBIM in combination would be a nice future perspective

**Comment by sudo-paras-shah** [comment]
Since this plugin allows you to **import** georeferenced rasters, it should also have the capability to **export** back rasters as well, right? Because I have added a Blender object into the scene with the elevation data and I want to export it back as a georeferenced raster to be viewable in GIS softwares such as QGIS and the present export option of a shapefile does not do well for me since I don't think it exports the geo data as well (The attribute table only has one field objId)

 I hope this falls under the same topic heading of additional GIS formats
